ta/avb/entry.c

```diff
@@ -212,9 +238,105 @@ static TEE_Result write_lock_state(uint32_t pt,
 	if (count == sizeof(lock_state) && lock_state == wlock_state)
 		goto out;
 
-	res = create_state(wlock_state, &h);
+	res = create_pobj((char *)rb_state_name, (char *)&wlock_state,
+			   sizeof(wlock_state), &h);
+out:
+	TEE_CloseObject(h);
+	return res;
+}
+
+static TEE_Result write_persist_value(uint32_t pt,
+				      TEE_Param params[TEE_NUM_PARAMS])
+{
+	const uint32_t exp_pt = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_MEMREF_INPUT,
+						TEE_PARAM_TYPE_NONE,
+						TEE_PARAM_TYPE_NONE);
+	TEE_Result res;
+	TEE_ObjectHandle h;
+	char name[TEE_OBJECT_ID_MAX_LEN];
+	char *value;
+	uint32_t value_sz;
+
+	if (pt != exp_pt)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	if (strlen(params[0].memref.buffer) >= TEE_OBJECT_ID_MAX_LEN ||
```

Comment:
There's no need to treat this as a null terminated string. In fact `strlen()` is dangerous to use here, `strnlen()` would have been better.

In order to avoid conflicts with the name "rb_state" you can just add a prefix, for instance "named_value_".

You should probably have a helper function to create a suitable object id based on the input.

Reply:
Due to AVB spec, it has be a null terminated string, as there is no size param, that indicates its size:
```
AvbIOResult (*read_persistent_value)(AvbOps* ops,
  const char* name,
  size_t buffer_size,
  uint8_t* out_buffer,
  size_t* out_num_bytes_read);
```